libnavigation-ui/src/main/java/com/mapbox/navigation/ui/feedback/FeedbackBottomSheet.java

```diff
@@ -287,8 +331,145 @@ public void onFeedbackItemClick(ImageView view, int feedbackPosition) {
         view.setPressed(!view.isPressed());
       }
       FeedbackItem feedbackItem = notificationIssueAdapter.getFeedbackItem(feedbackPosition);
+      onFeedbackSelected(feedbackItem);
+    }
+  };
+
+  private void onFeedbackSelected(FeedbackItem feedbackItem) {
+    if (feedbackFlowType == FEEDBACK_DETAIL_FLOW) {
+      launchDetailFlow(feedbackItem);
+    } else {
       feedbackBottomSheetListener.onFeedbackSelected(feedbackItem);
       startTimer();
     }
-  };
+  }
+
+  private void launchDetailFlow(@NonNull FeedbackItem feedbackItem) {
+    initCountDownAnimation();
+    selectedFeedbackItem = feedbackItem;
+
+    feedbackBottomSheetTitleText.setText(feedbackItem.getFeedbackText().replace('\n', ' '));
+    initFeedbackIssueDetailRecyclerView(feedbackItem);
+    feedbackMainLayout.setVisibility(View.GONE);
+    feedbackDescriptionsLayout.setVisibility(View.VISIBLE);
+  }
+
+  private void initFeedbackIssueDetailRecyclerView(FeedbackItem feedbackItem) {
+    feedbackDescriptionAdapter = new FeedbackDescriptionAdapter(
+      buildFeedbackIssueDetailList(feedbackItem), descriptionItemClickListener);
+    feedbackDescriptionItems.setAdapter(feedbackDescriptionAdapter);
+    feedbackDescriptionItems.setOverScrollMode(RecyclerView.OVER_SCROLL_ALWAYS);
+    feedbackDescriptionItems.setLayoutManager(new LinearLayoutManager(this.getContext()));
+  }
+
+  private FeedbackDescriptionAdapter.OnDescriptionItemClickListener descriptionItemClickListener =
+    new FeedbackDescriptionAdapter.OnDescriptionItemClickListener() {
+      @Override
+      public boolean onItemClick(int position) {
+        FeedbackDescriptionItem item = feedbackDescriptionAdapter.getFeedbackDescriptionItem(position);
+        if (selectedFeedbackItem.getFeedbackSubType().add(item.getFeedbackDescription())) {
+          item.setChecked(true);
+          return true;
+        } else {
+          selectedFeedbackItem.getFeedbackSubType().remove(item.getFeedbackDescription());
+          item.setChecked(false);
+          return false;
+        }
+      }
+    };
+
+  private List<FeedbackDescriptionItem> buildFeedbackIssueDetailList(FeedbackItem feedbackItem) {
+    List<FeedbackDescriptionItem> list = new ArrayList<>();
+    switch (feedbackItem.getFeedbackType()) {
```

Comment:
NIT I'm not a big fan of `switch`, could we refactor this long one into a map / dictionary?